qiita_db/support_files/patches/33.sql

```diff
@@ -0,0 +1,435 @@
+-- September 4, 2015
+-- Change the database structure to remove the RawData, PreprocessedData and
+-- ProcessedData division to unify it into the Artifact object
+
+-- Creating the new tables
+
+-- Rename the table filetype
+ALTER TABLE qiita.filetype RENAME TO artifact_type;
+ALTER TABLE qiita.artifact_type RENAME COLUMN filetype_id TO artifact_type_id;
+ALTER TABLE qiita.artifact_type RENAME COLUMN type TO artifact_type;
+ALTER TABLE qiita.artifact_type ADD description varchar;
+
+-- Rename the processed_data_status table
+ALTER TABLE qiita.processed_data_status RENAME TO visibility;
+ALTER TABLE qiita.visibility RENAME COLUMN processed_data_status_id TO visibility_id;
+ALTER TABLE qiita.visibility RENAME COLUMN processed_data_status TO visibility;
+ALTER TABLE qiita.visibility RENAME COLUMN processed_data_status_description TO visibility_description;
+UPDATE qiita.visibility
+    SET visibility_description = 'Only visible to the owner and shared users'
+    WHERE visibility = 'private';
+UPDATE qiita.visibility
+    SET visibility_description = 'Visible to everybody'
+    WHERE visibility = 'public';
+
+-- Artifact table - holds an abstract data object from the system
+CREATE TABLE qiita.artifact (
+    artifact_id                         bigserial  NOT NULL,
+    generated_timestamp                 timestamp  NOT NULL,
+    command_id                          bigint  ,
+    command_parameters_id               bigint  ,
+    visibility_id                       bigint  NOT NULL,
+    file_status                         varchar  NOT NULL,
+    artifact_type_id                    integer  ,
+    can_be_submitted_to_ebi             bool DEFAULT 'FALSE' NOT NULL,
+	can_be_submitted_to_vamps           bool DEFAULT 'FALSE' NOT NULL,
+    submitted_to_vamps                  bool DEFAULT 'FALSE' NOT NULL,
+    CONSTRAINT pk_artifact PRIMARY KEY ( artifact_id )
+ ) ;
+CREATE INDEX idx_artifact_0 ON qiita.artifact ( visibility_id ) ;
+CREATE INDEX idx_artifact_1 ON qiita.artifact ( artifact_type_id ) ;
+CREATE INDEX idx_artifact ON qiita.artifact ( command_id ) ;
+COMMENT ON TABLE qiita.artifact IS 'Represents data in the system';
+COMMENT ON COLUMN qiita.artifact.visibility_id IS 'If the artifact is sandbox, awaiting_for_approval, private or public';
+COMMENT ON COLUMN qiita.artifact.file_status IS 'If it is linking, unlinking or idle';
```

Comment:
That cannot be in the artifact_filepath table. However, your first comment is really good. The only reason we had the linking status was because we needed to hack at some point the creation of a raw data w/o files. I think we can remove this column and don't allow artifacts w/o files.